### PR TITLE
feat(issue_platform): Implement function to create an issue from an occurrence and event

### LINKED ISSUE #42052: Ingestion: Create/update issues based on IssueOccurrence info
Examine the fingerprint(s) passed with the `IssueOccurrence` and decide whether to create/update an issue based on it. `_save_aggregate_performance` could be a decent base here.

Note: This shouldn't happen as part of `EventManager.save` since saving the event is separate to ingesting the `IssueOccurrence`

### PR DESCRIPTION
**Pull request by wedamija**
This implements creating an issue from an occurrence, and does so when we save a new issue occurrence.

Some of the fields that we store on the issue are probably not 100% correct yet. We can iterate on these as we work on this more.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -2887,6 +2887,8 @@ def build_cdc_postgres_init_db_volume(settings):
 SENTRY_PERFORMANCE_ISSUES_RATE_LIMITER_OPTIONS = {}
 SENTRY_PERFORMANCE_ISSUES_REDUCE_NOISE = False
 
+SENTRY_ISSUE_PLATFORM_RATE_LIMITER_OPTIONS = {}
```
Comment (wedamija) [inline src/sentry/conf/server.py:2890]: https://github.com/getsentry/getsentry/pull/9080

File: src/sentry/issues/ingest.py [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
+from sentry.ratelimits.sliding_windows import Quota, RedisSlidingWindowRateLimiter, RequestedQuota
+from sentry.types.issues import GROUP_TYPE_TO_CATEGORY
+from sentry.utils import metrics
+
+issue_rate_limiter = RedisSlidingWindowRateLimiter(
+    **settings.SENTRY_ISSUE_PLATFORM_RATE_LIMITER_OPTIONS
+)
+# This should probably be configurable per team
+ISSUE_QUOTA = Quota(3600, 60, 5)
+GROUPHASH_IGNORE_LIMIT = 3
```
Comment (rjo100) [inline src/sentry/issues/ingest.py:31]: do we want to start with this on? feel like let's keep it off until GA, could keep it on for EA if that makes sense.

Comment (rjo100) [reply src/sentry/issues/ingest.py:31]: i see it's not implemented below so that's fine

File: src/sentry/issues/ingest.py [hunk 3]
```diff
@@ -13,6 +42,150 @@ def save_issue_occurrence(occurrence_data: IssueOccurrenceData, event: Event) ->
...
+    platform: Optional[str]
+    message: str
+    level: Optional[int]
+    culprit: str
+    last_seen: datetime
+    first_seen: datetime
+    active_at: datetime
+    type: int
+    data: OccurrenceMetadata
+    first_release: Optional[Release]
```
Comment (rjo100) [inline src/sentry/issues/ingest.py:69]: at this point we might want all the types to live in their own files

File: src/sentry/issues/ingest.py [hunk 4]
```diff
@@ -13,6 +42,150 @@ def save_issue_occurrence(occurrence_data: IssueOccurrenceData, event: Event) ->
...
+
+    event_type = get_event_type(event.data)
+    event_metadata = dict(event_type.get_metadata(event.data))
+    event_metadata = dict(event_metadata)
+    event_metadata["title"] = occurrence.issue_title
+
+    return {
+        "type": event_type.key,
+        # Not totally sure if this makes sense?
+        "culprit": occurrence.subtitle,
```
Comment (rjo100) [inline src/sentry/issues/ingest.py:118]: i guess this makes sense. would probably be too much trouble to involve `IssueEvidence`. what do we show with our integrations currently that subtitle is replacing/substituting for?

File: src/sentry/issues/ingest.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/issues/ingest.py:69]: This type is specific to this one file., I don't think we need to move it out really. Tbh this specific type might be overkill, although it seems nicer than just using `Mapping[str, Any]`

File: src/sentry/issues/ingest.py (hunk 4 above)
Comment (wedamija) [reply src/sentry/issues/ingest.py:118]: My memory is a bit vague on how the frontend displays things here on the issue list, but from what I remember `title` is the main title, `location` is the path and `culprit` is the subtitle. Tbh I'm not worried about getting this perfect right now, we can write this data here and modify it later if necessary. Comment was more a note to revisit it later if necessary.

File: src/sentry/issues/ingest.py (hunk 3 above)
Comment (rjo100) [reply src/sentry/issues/ingest.py:69]: makes sense. i just feel like we're at 3+ TypedDicts for this stuff now haha between occurrences and events and whatnot

**Review by rjo100 — APPROVED** [review]

**Pull request merged**